deployment/ccip/changeset/cs_deploy_token_pools.go

```diff
@@ -0,0 +1,215 @@
+	}
+
+	// Validate acceptLiquidity based on requested pool type
+	if i.Type == LockReleaseTokenPool && i.AcceptLiquidity == nil {
+		return errors.New("accept liquidity must be defined for lock release pools")
+	}
+	if i.Type != LockReleaseTokenPool && i.AcceptLiquidity != nil {
+		return errors.New("accept liquidity must be nil for burn mint pools")
+	}
+
+	// Regardless of requested type, we should check if a token pool of any type already exists
+	tokenPools, err := getAllTokenPoolsWithSymbolAndVersion(state, chain.Client, tokenSymbol, currentTokenPoolVersion)
+	if err != nil {
+		return fmt.Errorf("failed to get all token pools with symbol %s on chain %s: %w", tokenSymbol, chain.String(), err)
+	}
+	if len(tokenPools) > 0 && !i.ForceDeployment {
+		return fmt.Errorf("token pool with version %s already exists for %s on %s (use forceDeployment to bypass)", currentTokenPoolVersion, tokenSymbol, chain.String())
+	}
+
+	return nil
+}
+
+// DeployTokenPoolContractsConfig defines the token pool contracts that need to be deployed on each chain.
+type DeployTokenPoolContractsConfig struct {
+	// Symbol is the symbol of the token for which we are deploying a pool.
+	TokenSymbol TokenSymbol
+	// NewPools defines the per-chain configuration of each new pool
+	NewPools map[uint64]DeployTokenPoolInput
+}
+
+func (c DeployTokenPoolContractsConfig) Validate(env deployment.Environment) error {
+	// Ensure that required fields are populated
+	if c.TokenSymbol == TokenSymbol("") {
+		return errors.New("token symbol must be defined")
+	}
+
+	state, err := LoadOnchainState(env)
+	if err != nil {
+		return fmt.Errorf("failed to load onchain state: %w", err)
+	}
+	for chainSelector, poolConfig := range c.NewPools {
+		err := deployment.IsValidChainSelector(chainSelector)
+		if err != nil {
+			return fmt.Errorf("failed to validate chain selector %d: %w", chainSelector, err)
+		}
+		chain, ok := env.Chains[chainSelector]
+		if !ok {
+			return fmt.Errorf("chain with selector %d does not exist in environment", chainSelector)
+		}
+		chainState, ok := state.Chains[chainSelector]
+		if !ok {
+			return fmt.Errorf("chain with selector %d does not exist in state", chainSelector)
+		}
+		if router := chainState.Router; router == nil {
+			return fmt.Errorf("missing router on %s", chain.String())
+		}
+		if rmnProxy := chainState.RMNProxy; rmnProxy == nil {
+			return fmt.Errorf("missing rmnProxy on %s", chain.String())
+		}
+		err = poolConfig.Validate(chain, chainState, c.TokenSymbol)
+		if err != nil {
+			return fmt.Errorf("failed to validate token pool config for chain selector %d: %w", chainSelector, err)
+		}
+	}
+	return nil
+}
+
+// DeployTokenPoolContractsChangeset deploys new pools for a given token across multiple chains.
+func DeployTokenPoolContractsChangeset(env deployment.Environment, c DeployTokenPoolContractsConfig) (deployment.ChangesetOutput, error) {
```

Comment:
Do we want to support MCMS for this changeset? Also would be benefit from simulation in the DeployerGroup even if we don't use MCMS?

Reply:
MCMS doesn't apply to this one, since it just involves deploying contracts